### DISCUSSION TITLE
How can I fix the built-in speaker in hackintosh 13.1

### DISCUSSION THREAD

**Discussion by abudibel** [Qonfused/ASUS-ZenBook-Duo-14-UX481-Hackintosh#17] [Post-Install Help] [answered]
How can I fix the built-in speaker in hackintosh 13.1 
me laptop is a dell inspiron 3593 i7 1065g7 intel iris plus nvidia mx230(not working) 16gb ram
and this is the efi (https://drive.google.com/drive/folders/1H-4w7JJahG7pIg1uxsXjbTtYplW-yXWB?usp=share_link)
can I get it fixed?
and the touch pad is acting like a mouse how can I lest work as a touch pad?

**Comment by Maleficent-Magik** [comment]
Uuh, excuse me? You're on ASUS ZENBOOK DUO UX481... Not Dell Inspiron laptop seriously...

Please, don't spam on other project. Please ask on /r/Hackintosh on Reddit or Discord. Thanks.

@Qonfused don't accept this request... And me too I don't accept this request '

**Comment by Qonfused** [comment, marked as answer]
A better place to ask for help with your laptop would be in the [Hackintosh Paradise Discord](https://discord.gg/u8V7N5C) or on https://www.reddit.com/r/hackintosh/.

Here are a couple things you could try or look into:

* If you've tried every layout-id for your audio codec, you can add SSDT-HPET.aml (make using [SSDTTime](https://github.com/corpnewt/SSDTTime)) and use AppleALC instead. [Make sure you've found the right codec and tried all layout-ids](https://dortania.github.io/OpenCore-Post-Install/universal/audio.html) before trying this.

* You'll need to disable the VoodooPS2 mouse and trackpad plugin kexts and add `-vi2c-force-polling` to your boot args to [start debugging your trackpad](https://dortania.github.io/Getting-Started-With-ACPI/Laptops/trackpad.html) (assuming it's I2C). Your trackpad should be handled by VoodooI2C/VoodooRMI/etc (depending on what interface it uses).

However your EFI folder looks very problematic to debug. It'd be better to start with [the dortania install guide](https://dortania.github.io/OpenCore-Install-Guide/) ([this is the **Ice Lake** section for setting up your config.plist](https://dortania.github.io/OpenCore-Install-Guide/config-laptop.plist/icelake.html#starting-point)); you'll end up with a more functional system this way with fewer hurdles to jump through.

An important thing to note is that you have an Ice Lake laptop with Iris Plus graphics (which is supported), but your MX230 is not supported and will need to be disabled in macOS.

I'd defer to either of the two links (Discord or Reddit) after walking through the install guide; no support is given there for pre-built EFIs (and it'd be very difficult to debug in its current state).
